### PR TITLE
chore(rust/gui-client): report account slug to Sentry

### LINKED ISSUE #7087: chore(rust/gui-client): parse Firezone slug from deep link URL and send it through Sentry

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #7087

image

### CODE REVIEW

File: rust/gui-client/src-common/src/auth.rs [hunk 1]
```diff
@@ -4,32 +4,35 @@ use anyhow::Result;
...
-use std::path::PathBuf;
+use serde::{Deserialize, Serialize};
+use std::path::{Path, PathBuf};
 use subtle::ConstantTimeEq;
 use url::Url;
 
 const NONCE_LENGTH: usize = 32;
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
```
Comment (ReactorScram) [inline rust/gui-client/src-common/src/auth.rs:15]: I could get rid of this and make it `anyhow` in a future PR

File: rust/gui-client/src-common/src/auth.rs [hunk 2]
```diff
@@ -61,14 +64,17 @@ impl Request {
...
+    pub(crate) state: SecretString,
 }
 
-pub struct Session {
-    pub actor_name: String,
+#[derive(Default, Deserialize, Serialize)]
+pub(crate) struct Session {
+    pub(crate) account_slug: String,
+    pub(crate) actor_name: String,
 }
```
Comment (ReactorScram) [inline rust/gui-client/src-common/src/auth.rs:78]: Turns out these didn't need to be `pub` but that's hard to lint for

File: rust/gui-client/src-common/src/auth.rs [hunk 3]
```diff
@@ -205,14 +213,26 @@ impl Auth {
...
             // In that case we just say the app is signed out
             Err(e) if e.kind() == std::io::ErrorKind::NotFound => return Ok(None),
-            Err(e) => return Err(Error::ReadActorName(e)),
+            Err(e) => return Err(Error::ReadFile(e)),
         };
+        match std::fs::read_to_string(session_data_path()?) {
+            Ok(x) => {
+                session = serde_json::from_str(&x).map_err(|_| Error::Serde)?;
+                firezone_telemetry::Hub::main()
+                    .configure_scope(|scope| scope.set_tag("account_slug", &session.account_slug));
```
Comment (thomaseizinger) [inline rust/gui-client/src-common/src/auth.rs]: Is there any value in setting this as part of the "user" concept that Sentry has? https://docs.rs/sentry/latest/sentry/struct.Scope.html#method.set_user

File: rust/gui-client/src-common/src/auth.rs [hunk 4]
```diff
@@ -205,14 +213,26 @@ impl Auth {
...
             // even though the token is in the cred manager.
             // In that case we just say the app is signed out
             Err(e) if e.kind() == std::io::ErrorKind::NotFound => return Ok(None),
-            Err(e) => return Err(Error::ReadActorName(e)),
+            Err(e) => return Err(Error::ReadFile(e)),
         };
+        match std::fs::read_to_string(session_data_path()?) {
+            Ok(x) => {
+                session = serde_json::from_str(&x).map_err(|_| Error::Serde)?;
+                firezone_telemetry::Hub::main()
```
Comment (thomaseizinger) [inline rust/gui-client/src-common/src/auth.rs]: Should there be a shortcut method on the telemetry module that just accepts the account-slug?

File: rust/gui-client/src-common/src/deep_link.rs [hunk 5]
```diff
@@ -48,12 +51,20 @@ pub fn parse_auth_callback(url_secret: &SecretString) -> Result<auth::Response>
         _ => bail!("URL path should be `/` or empty"),
     }
 
+    let mut account_slug = None;
     let mut actor_name = None;
     let mut fragment = None;
     let mut state = None;
 
+    // There's probably a way to get serde to do this
```
Comment (thomaseizinger) [inline rust/gui-client/src-common/src/deep_link.rs:59]: You could use an `iter().find_map` too which might be more concise?

**Review by thomaseizinger — APPROVED** [review]
Great work!

**Pull request merged**